Sources/MapboxMaps/Foundation/Extensions/Core/DownloadStatus.swift

```diff
@@ -0,0 +1,122 @@
+
+        let expected: Expected<AnyObject, AnyObject>
+        switch httpResult {
+        case let .success(response):
+            expected = Expected(value: response)
+        case let .failure(error):
+            expected = Expected(error: error)
+        }
+
+        self.init(downloadId: downloadId,
+                  state: state,
+                  error: error,
+                  totalBytes: totalBytes?.NSNumber,
+                  receivedBytes: receivedBytes,
+                  transferredBytes: transferredBytes,
+                  downloadOptions: downloadOptions,
+                  httpResult: expected)
+    }
+
+    /// Convenience to initialize a `DownloadStatus` when the download state is `.pending`
+    ///
+    /// - Parameters:
+    ///   - error: Contains error information in case of failure when state is
+    ///         set to `DownloadState.failed`.
+    ///   - totalBytes: Total amount of bytes to receive. In some cases this
+    ///         value is unknown until we get final part of the file.
+    ///   - downloadOptions: Download options used to send the download request.
+    ///   - httpResult: HTTP result. This field is only set for `DownloadState.failed`
+    ///         and `DownloadState.finished`.
+    ///         For `.failed` expect `HttpRequestError` to be provided for cases when
+    ///         `DownloadErrorCode` is `NetworkError`. For `.finished` `HttpResponseData`
+    ///         is set, but with empty data field (since all the data was written
+    ///         to the disk).
+    public convenience init(error: DownloadError?,
+                            totalBytes: UInt64?,
+                            downloadOptions: DownloadOptions,
+                            httpResult: Result<HttpResponseData, HttpRequestError>) {
+
+        let expected: Expected<AnyObject, AnyObject>
+        switch httpResult {
+        case let .success(response):
+            expected = Expected(value: response)
+        case let .failure(error):
+            expected = Expected(error: error)
+        }
+
+        self.init(error: error,
+                  totalBytes: totalBytes?.NSNumber,
+                  downloadOptions: downloadOptions,
+                  httpResult: expected)
+    }
+
+    /// HTTP result. This field is only set for `DownloadState.failed` and
+    /// `DownloadState.finished`.
+    ///
+    /// For `.failed` expect `HttpRequestError` to be provided for cases when
+    /// `DownloadErrorCode` is `NetworkError`.
+    ///  And for `.finished` `HttpResponseData` is set, but with empty data field
+    ///  (since all the data was written to the disk).
+    public var httpResult: Result<HttpResponseData, HttpRequestError>? {
+        guard let httpExpected = __httpResult else {
+            return nil
+        }
+
+        guard let expected = httpExpected as? Expected<HttpResponseData, HttpRequestError>  else {
+            fatalError("Invalid Expected types or none.")
```

Comment:
I think this would only happen due to `Invalid Expected types`. The `none` case would be caught be the previous guard statement, right?